samtranslator/model/eventsources/push.py

```diff
@@ -657,6 +659,15 @@ def _add_swagger_integration(self, api, function, intrinsics_resolver):
                             ),
                         )
 
+                    if not isinstance(method_authorizer, str):
+                        raise InvalidEventException(
+                            self.relative_id,
+                            "Unable to set Authorizer [{authorizer}] on API method [{method}] for path [{path}] "
```

Comment:
More of a general note, but it seems this whole function could benefit from refactoring. It's large, repetitive and consists of multiple intertwined responsibilities (validation and processing).

Reply:
Thats a good point and I totally agree. however it feels like a different task as this refactoring may be needed at several places, not just this function.

Reply:
Yep, agreed! Meant it as a general call-out; PR itself looks good (forgot to approve!).